index.js

```diff
@@ -443,8 +443,28 @@ function callHandlers(handlers, args, method) {
   for (var i = handlers.length - 1; i >= 0; i -= 1) {
     method = handlers[i].bind(null, method, args);
   }
+
+  //make sure the handler chain is not started, before the corresponding
+  //pouchdb instance is fully initialized. Otherwise it might result
+  //in a double calling the augmented methods
+  var promise;
+  if (!args.base.taskqueue.isReady) {
+    promise = new Promise(function (resolve, reject) {
+      args.base.taskqueue.addTask(function (failed) {
+        if (failed) {
+          reject(failed);
+        } else {
+          resolve();
+        }
+      });
+    }).then(function () {
+      return method();
+    });
+  } else {
+    promise = method();
+  }
```

Comment:
You are right. I usually use this pattern as well. I have got no idea, why I didn't do it here. Should I change it and push a new commit, or do you want to stay with the original implementation?